ambry-store/src/main/java/com.github.ambry.store/LogSegmentNameHelper.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Comparator;
+
+
+/**
+ * Helper for working with log segment names.
+ * <p/>
+ * Log segments will have a name "pos_gen" where pos represents their relative position and "gen" represents the
+ * generation number of the log segment at "pos".
+ * <p/>
+ * The file name is a combination of the segment name and a suffix "_log"
+ */
+class LogSegmentNameHelper {
```

Comment:
This is actually meant to take versions - it hasn't been added because it is not required here but the class can be extended to handle versioning - so i don't want to limit its scope.